test/spec/utils_spec.js

```diff
@@ -1032,6 +1032,32 @@ describe('Utils', function () {
         }
       });
     });
+
+    it('properly merge objects with undefined property types', function() {
+      const object1 = {
+        depth0_A: {
+          depth1_A: ['a', 'b', 'c'],
+          depth1_B: 'abc',
+          depth1_C: 123
+        }
+      };
+      const object2 = {
+        depth0_A: {
+          depth1_A: undefined,
+          depth1_D: true,
+        }
+      };
+
+      const result = utils.mergeDeep({}, object1, object2);
+      expect(result).to.deep.equal({
+        depth0_A: {
+          depth1_A: ['a', 'b', 'c'],
```

Comment:
I can see an argument for either way - it'd also make sense to me if this was undefined. In other instances where the values are not objects and different, the source's overrides the target's. 

In a vacuum then I'd prefer to keep the status quo, since it carries less risk. What prompted this change?

Reply:
We have a publisher where we see the user.eids in the bidrequest getting overwritten by undefined when the fpd data has user.eids value as 'undefined'.
While the fix solves this problem, if there is a possibility that use cases from other bidders that might result into unexpected output, or we want to maintain the existing behaviour, we can move the fix into our bidder.
Please advice.

Reply:
> when the fpd data has user.eids value as 'undefined'.

could it be that is the publishers' intention? If so, this would be breaking

Reply:
instead of putting this in your adapter, it seems safer if the eid merge used the alternative function, or you add an argument to the existing function (skipUndefined), default it to the burrent behavior, and change only the invocation of mergeDeep you want to have the new behavior.

Is the invocation you want to affect of mergeDeep in your adapter or in core?

Reply:
is it this one? https://github.com/prebid/Prebid.js/blob/499933951b0b0cba5a1472c9e94ebbaee2726d09/modules/pubmaticBidAdapter.js#L1303

Reply:
To elaborate on cleaning out undefineds from FPD: it could be done by wrapping https://github.com/prebid/Prebid.js/blob/499933951b0b0cba5a1472c9e94ebbaee2726d09/src/prebid.js#L562-L565

in `JSON.parse(JSON.stringify(`  + falling back to current behavior on error.

Reply:
> is it this one?
> 
> https://github.com/prebid/Prebid.js/blob/499933951b0b0cba5a1472c9e94ebbaee2726d09/modules/pubmaticBidAdapter.js#L1303

Yes thats right.

Reply:
> > when the fpd data has user.eids value as 'undefined'.
> 
> could it be that is the publishers' intention? If so, this would be breaking. Particularly if they are doing this in other parts of their fpd objects.

This is not intentional. The publisher raised an issue where IDs although generated aer not getting sent to PubMatic call.